feature/experimental/isis/otg_tests/isis_change_lsp_lifetime_test/isis_change_lsp_lifetime_test.go

```diff
@@ -0,0 +1,241 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package isis_change_lsp_lifetime_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/openconfig/featureprofiles/feature/experimental/isis/otg_tests/internal/session"
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/featureprofiles/internal/otgutils"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygnmi/ygnmi"
+	"github.com/openconfig/ygot/ygot"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+const (
+	lspLifetime = 500
+	v4Route1    = "203.0.113.0"
+	v6Route1    = "2001:db8::203:0:113:0"
+	v4Route     = "203.0.113.0/30"
+	v6Route     = "2001:db8::203:0:113:0/126"
+	v4IP        = "203.0.113.1"
+	v6IP        = "2001:db8::203:0:113:1"
+	v4NetName   = "isisv4Net"
+	v6NetName   = "isisv6Net"
+	v4FlowName  = "v4Flow"
+	v6FlowName  = "v6Flow"
+)
+
+// configureISIS configures isis on DUT.
+func configureISIS(t *testing.T, ts *session.TestSession) {
+	t.Helper()
+	d := ts.DUTConf
+	netInstance := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(ts.DUT))
+	prot := netInstance.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS, session.ISISName)
+	prot.Enabled = ygot.Bool(true)
+
+	isis := prot.GetOrCreateIsis()
+	globalIsis := isis.GetOrCreateGlobal()
+
+	// Global configs
+	globalIsis.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV4, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
+	globalIsis.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV6, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
+	globalIsis.LevelCapability = oc.Isis_LevelType_LEVEL_2
+	globalIsis.GetOrCreateTimers().LspLifetimeInterval = ygot.Uint16(lspLifetime)
+}
+
+// configureOTG configures isis and traffic on OTG.
+func configureOTG(t *testing.T, ts *session.TestSession) {
+	t.Helper()
+
+	// netv4 is a simulated network containing the ipv4 addresses specified by targetNetwork
+	netv4 := ts.ATEIntf1.Isis().V4Routes().Add().SetName(v4NetName).SetLinkMetric(10)
+	netv4.Addresses().Add().SetAddress(v4Route1).SetPrefix(uint32(session.ATEISISAttrs.IPv4Len))
+
+	// netv6 is a simulated network containing the ipv6 addresses specified by targetNetwork
+	netv6 := ts.ATEIntf1.Isis().V6Routes().Add().SetName(v6NetName).SetLinkMetric(10)
+	netv6.Addresses().Add().SetAddress(v6Route1).SetPrefix(uint32(session.ATEISISAttrs.IPv6Len))
+
+	// We generate traffic entering along port2 and destined for port1
+	srcIntf := ts.MustATEInterface(t, session.ATETrafficAttrs.Name)
```

Comment:
Is srcIntf port2?

You can directly use `ts.ATEIntf2`

Reply:
It is created by default already.

Reply:
modified